nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -1476,38 +1498,48 @@ Nav2Panel::getGoalStatusLabel(int8_t status)
 }
 
 inline QString
-Nav2Panel::getNavToPoseFeedbackLabel(nav2_msgs::action::NavigateToPose::Feedback msg)
+Nav2Panel::getNavToPoseFeedbackLabel(const nav2_msgs::action::NavigateToPose::Feedback msg)
 {
   return QString(std::string("<table>" + toLabel(msg) + "</table>").c_str());
 }
 
 inline QString
-Nav2Panel::getNavThroughPosesFeedbackLabel(nav2_msgs::action::NavigateThroughPoses::Feedback msg)
+Nav2Panel::getNavThroughPosesFeedbackLabel(const nav2_msgs::action::NavigateThroughPoses::Feedback msg)
 {
   return QString(
     std::string(
-      "<table><tr><td width=150>Poses remaining:</td><td>" +
+      "<table><tr><td width=200>Poses remaining:</td><td>" +
       std::to_string(msg.number_of_poses_remaining) +
       "</td></tr>" + toLabel(msg) + "</table>").c_str());
 }
 
+inline QString
+Nav2Panel::getFollowWaypointsFeedbackLabel(const nav2_msgs::action::FollowWaypoints::Feedback msg)
+{
+  return QString(
+    std::string(
+      "<table><tr><td width=200>Total Time Taken:</td><td>" +
+      toString(rclcpp::Duration(msg.elapsed_time).seconds(), 2) + " s"
+      "</td></tr></table>").c_str());
+}
+
 template<typename T>
-inline std::string Nav2Panel::toLabel(T & msg)
+inline std::string Nav2Panel::toLabel(const T & msg)
 {
   return std::string(
-    "<tr><td width=150>ETA:</td><td>" +
+    "<tr><td width=200>ETA:</td><td>" +
     toString(rclcpp::Duration(msg.estimated_time_remaining).seconds(), 0) + " s"
-    "</td></tr><tr><td width=150>Distance remaining:</td><td>" +
+    "</td></tr><tr><td width=200>Distance remaining:</td><td>" +
     toString(msg.distance_remaining, 2) + " m"
-    "</td></tr><tr><td width=150>Time taken:</td><td>" +
+    "</td></tr><tr><td width=200>Time Taken for Current WP:</td><td>" +
```

Comment:
Is this just for WPF or applies to other things?

Reply:
It applies to all since this code is used for all. Do you want me to make it conditionally display `for Current WP` and `for current Pose` or something instead?

Reply:
I don't want any changes to the normal behavior, only for the WP follower :smile: